tests/components/rainbird/test_binary_sensor.py

```diff
@@ -38,3 +40,33 @@ async def test_rainsensor(
         "friendly_name": "Rain Bird Controller Rainsensor",
         "icon": "mdi:water",
     }
+
+    entity = entity_registry.async_get("binary_sensor.rain_bird_controller_rainsensor")
+    assert entity
+    assert entity.unique_id == "1263613994342-rainsensor"
+
+
+@pytest.mark.parametrize(
+    ("config_entry_unique_id"),
+    [
+        (None),
+    ],
+)
+async def test_no_unique_id(
+    hass: HomeAssistant,
+    setup_integration: ComponentSetup,
+    responses: list[AiohttpClientMockResponse],
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test rainsensor binary sensor with no unique id."""
+
+    assert await setup_integration()
+
+    rainsensor = hass.states.get("binary_sensor.rain_bird_controller_rainsensor")
+    assert rainsensor is not None
+    assert (
+        rainsensor.attributes.get("friendly_name") == "Rain Bird Controller Rainsensor"
+    )
+
+    entity = entity_registry.async_get("binary_sensor.rain_bird_controller_rainsensor")
```

Comment:
```suggestion
    entity_entry = entity_registry.async_get("binary_sensor.rain_bird_controller_rainsensor")
```

Reply:
Renamed all other instances of this as well, will merge when CI passes.